modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/MatchOnlyTextFieldMapper.java

```diff
@@ -350,16 +353,16 @@ protected BytesRef storedToBytesRef(Object stored) {
                 };
             }
             return new SourceValueFetcherSortedBinaryIndexFieldData.Builder(
-                name(),
+                concreteFieldName(),
```

Comment:
here too, if it has to do with source access, not sure we should use the lucene field name.

Reply:
Corrected in all places we access source! Thanks